firmware/quadruna/VC/src/tasks.c

```diff
@@ -475,6 +515,44 @@ _Noreturn void tasks_runCanRx(void)
     }
 }
 
+_Noreturn void tasks_runLogging(void)
+{
+    static uint32_t count = 0;
+    for (;;)
+    {
+        if (!sd_functional())
+        {
+            osDelay(1000);
+            continue;
+        }
+
+        io_canLogging_recordMsgFromQueue();
+        count++;
+        write_count++;
+        if (count > 256)
+        {
+            io_canLogging_sync();
+            count = 0;
+        }
+    }
+}
+
+static void can_msg_received_callback(CanMsg *rx_msg)
```

Comment:
this is an io_can construct??

Reply:
This callback function gets called when rx can interrupt fire. Originally, we only called `io_can_msgReceivedCallback` which push the message to a queue, but now we need to push the message to another queue for logging. So I need to replace the behavior of original callback function to this new callback function.